analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSyntaxHelper.cs

```diff
@@ -327,8 +327,8 @@ public static SimpleNameSyntax GetIdentifier(this ExpressionSyntax expression)
             expression switch
             {
                 MemberAccessExpressionSyntax memberAccess => memberAccess.Name.Identifier.ValueText,
-                IdentifierNameSyntax identifierName => identifierName.Identifier.ValueText,
-                GenericNameSyntax genericNameSyntax => genericNameSyntax.Identifier.ValueText,
+                SimpleNameSyntax simpleNameSyntax => simpleNameSyntax.Identifier.ValueText,
```

Comment:
`SimpleNameSyntax` is the base class for both `IdentifierNameSyntax` and `GenericNameSyntax`.